src/compiler/parser.ts

```diff
@@ -1407,7 +1407,51 @@ namespace Parser {
 
     // Share a single scanner across all calls to parse a source file.  This helps speed things
     // up by avoiding the cost of creating/compiling scanners over and over again.
-    var scanner = createScanner(ScriptTarget.Latest, /*skipTrivia*/ true);
+    var {
+        clearCommentDirectives,
+        getCommentDirectives,
+        getNumericLiteralFlags,
+        getStartPos,
+        getTextPos,
+        getToken,
+        getTokenFlags,
+        getTokenPos,
+        getTokenPos,
+        getTokenText,
+        getTokenValue,
+        hasExtendedUnicodeEscape,
+        hasExtendedUnicodeEscape: scanner_hasExtendedUnicodeEscape,

```

Comment:
Is it bad that I wish these all were `scanner_` prefixed? Maybe it's not such a big deal to remember where these come from, I suppose.